src/librbd/ImageCtx.h

```diff
@@ -250,7 +250,7 @@ namespace librbd {
 		  cls::rbd::SnapshotNamespace in_snap_namespace,
 		  librados::snap_t id,
 		  uint64_t in_size, parent_info parent,
-                  uint8_t protection_status, uint64_t flags);
+                  uint8_t protection_status, uint64_t flags, utime_t timestamp = utime_t());
```

Comment:
This is the issue we really need discussion... this func "add_snap" will be called in two places:
1) for snap list command, if there are 3 snapshots existed for an image, add_snap will be called one by one. And this is the case what we do in this pr.
2) create snap command. It will also call this add_snap, to add new snap shot info into image context. However, this new added snap shot info won't be used in any place. The list command will alway fetch it from backend each time... So I add a default value for timestamp for this "create snap" case. In this way, I don't need to regolden the test cases. Another way is: don't call this add_snap when create snapshot, since new snap shot info won't be used.

What is your opinion? forgive my pool english expression...